py/server/deephaven/stream/table_publisher.py

```diff
@@ -0,0 +1,123 @@
+_JRuntimeException = jpy.get_type("java.lang.RuntimeException")
+
+
+class TablePublisher(JObjectWrapper):
+    """Supports publishing Table data."""
+
+    j_object_type = _JTablePublisher
+
+    def __init__(self, j_table_publisher: jpy.JType):
+        self.j_table_publisher = j_table_publisher
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_table_publisher
+
+    @property
+    def is_alive(self) -> bool:
+        """Checks whether this publisher is alive; if False, the publisher should stop publishing new data and release
+        any related resources as soon as practicable since publishing won't have any downstream effects.
+
+        Once this is False, it will always remain False. For more prompt notifications, publishers may
+        prefer to use on_shutdown_callback during construction.
+
+        Returns:
+            if this publisher is alive
+        """
+        return self.j_table_publisher.isAlive()
+
+    def add(self, table: Table):
+        """Adds a snapshot of the data from table into the blink table according to the blink table's definition.
+
+        All of the data from table will be:
+
+        1. consistent with a point in time
+        2. fully contained in a single blink table's update cycle
+        3. non-interleaved with any other calls to add (concurrent, or not)
+
+        Args:
+            table (Table): the table to add
+        """
+        self.j_table_publisher.add(table.j_table)
+
+    def publish_failure(self, failure: Exception):
+        """Publish a failure for notification to the listeners of the blink table. Future calls to add will silently
+        return. Will cause the on-shutdown callback to be invoked if it hasn't already been invoked.
+
+        Args:
+            failure (Exception): the failure
+        """
+        self.j_table_publisher.publishFailure(_JRuntimeException(str(failure)))
+
+
+# Note: we may want to expand the column_definitions typing here and elsewhere in the future to also accept List[Column]
+
+
+def table_publisher(
+    name: str,
+    col_defs: Dict[str, DType],
+    on_shutdown_callback: Optional[Callable[[], None]] = None,
+    update_graph: Optional[UpdateGraph] = None,
+    chunk_size: int = 2048,
+) -> tuple[Table, TablePublisher]:
+    """Constructs a blink Table and TablePublisher.
+
+    The on_shutdown_callback, if present, is called one time when the publisher should stop publishing new data
+    and release any related resources as soon as practicable since publishing won't have any downstream effects.
+
+    The update_graph is the update graph that the resulting table will belong to. If unset, the update graph of
+    the current execution context will be used.
+
+    The chunk_size is the size at which chunks will be filled from the source table during an add.
+
+    The caller is responsible for keeping a reference to the resulting blink (or derived) Table.
```

Comment:
I would drop this line about the caller being responsible since that is generally expected in python.